extensions/amp-lightbox-gallery/1.0/provider.js

```diff
@@ -54,7 +54,15 @@ const CAPTION_PROPS = {
  * @return {PreactDef.Renderable}
  */
 export function LightboxGalleryProviderWithRef(
-  {children, onAfterClose, onAfterOpen, onBeforeOpen, render},
+  {
+    children,
+    onAfterClose,
+    onAfterOpen,
+    onBeforeOpen,
```

Comment:
FMI/suspicion: What is `onAfterOpen` for? Why do we need it? I'm suspicious of having both onBeforeOpen/onAfterOpen

Reply:
`onBeforeOpen` and `onAfterOpen` accounts for the animation timing discrepancy between them. Certain things have to happen before animation (such as unhiding the transparent custom element layer) and certain things have to happen after (such as calling `setAsContainer` on an element that only exists after the animation sequence triggers its render).